ui_testing/pages/header_page.py

```diff
@@ -220,6 +222,177 @@ def click_on_clear_all(self):
+
+    def get_role_name(self):
+            return self.base_selenium.get_text(element='roles_and_permissions:role_name').split('\n')[0]
+
+    def create_new_role(self, sleep=True, role_name = ''):
+        self.base_selenium.LOGGER.info(' + Create new role.')
+        self.base_selenium.click(element='roles_and_permissions:new_role_btn')
+        time.sleep(self.base_selenium.TIME_SMALL)
+        role_name = self.set_role_name(role_name)
+
+        role_data = {
+            "role_name":role_name,
+        }
+        self.save(sleep)
+        return role_data
+
+    def archive_selected_roles(self):
+            self.base_selenium.scroll()
+            self.base_selenium.click(element='roles_and_permissions:right_menu')
+            self.base_selenium.click(element='roles_and_permissions:archive')
+            self.confirm_popup()
+
+    def get_archived_roles(self):
+        self.base_selenium.scroll()
+        self.base_selenium.click(element='roles_and_permissions:right_menu')
+        self.base_selenium.click(element='roles_and_permissions:archived')
+        self.sleep_small()
+
+    def click_on_role_right_menu(self):
+        self.base_selenium.LOGGER.info('Press on the right menu')
+        self.base_selenium.click(element='roles_and_permissions:right_menu')
+        self.sleep_small()
+
+    def click_on_role_delete_btn(self):
+        self.base_selenium.LOGGER.info('Press on the delete button')
+        self.base_selenium.click(element='roles_and_permissions:delete')
+        self.sleep_small()
+
+    def clear_role_name(self):
+        self.base_selenium.LOGGER.info('Clear role name')
+        self.base_selenium.clear_text(element='roles_and_permissions:role_name')
+
+    def click_on_pagination_page(self):
+        self.base_selenium.LOGGER.info('click on the pagination page')
+        self.base_selenium.click(element='roles_and_permissions:pagination_page')
+        self.sleep_small()
+
+    def click_on_master_data_permissions(self):
+        self.base_selenium.LOGGER.info('checked master data permissions')
+        self.base_selenium.click(element='roles_and_permissions:master_data_view_permissions')
+        self.base_selenium.click(element='roles_and_permissions:master_data_edit_permissions')
+        self.sleep_small()
+
+    def create_role_with_mater_data_permissions(self, sleep=True, role_name=''):
+        self.base_selenium.LOGGER.info(' + Create new role.')
+        self.base_selenium.click(element='roles_and_permissions:new_role_btn')
+        time.sleep(self.base_selenium.TIME_SMALL)
+        role_name = self.set_role_name(role_name)
+        self.click_on_master_data_permissions()
+
+        role_data = {
+            "role_name": role_name,
+
+        }
+        self.save(sleep)
+        return role_data
+
+    def click_on_logout_button(self):
+        self.base_selenium.LOGGER.info('Press on logout button')
+        self.base_selenium.click(element='login:logout_btn')
+        self.sleep_small()
```

Comment:
rename to logout, and move to login page

Reply:
Done